torchrl/collectors/distributed/distributed_submitit.py

```diff
@@ -0,0 +1,281 @@
+import os
+import socket
+import time
+from typing import OrderedDict
+
+SUBMITIT_ERR = None
+try:
+    import submitit
+
+    _has_submitit = False
+except ModuleNotFoundError as err:
+    _has_submitit = False
+    SUBMITIT_ERR = err
+import torch.cuda
+
+from torch.distributed import rpc
+
+from torchrl.collectors import MultiaSyncDataCollector
+from torchrl.collectors.collectors import (
+    _DataCollector,
+    MultiSyncDataCollector,
+    SyncDataCollector,
+)
+from torchrl.envs import EnvBase, EnvCreator
+
+SLEEP_INTERVAL = 1e-6
+MAX_TIME_TO_CONNECT = 1000
+DEFAULT_SLURM_CONF = {
+    "timeout_min": 10,
+    "slurm_partition": "train",
+    "slurm_cpus_per_task": 32,
+    "slurm_gpus_per_node": 0,
+}
+
+
+def init_collection_node(rank, rank0_ip, world_size):
+    """Sets up RPC on the distant node.
+
+    Args:
+        rank (int): the rank of the process;
+        rank0_ip (str): the IP address of the master process (rank 0)
+        world_size (int): the total number of nodes, including master.
+
+    """
+    os.environ["MASTER_ADDR"] = str(rank0_ip)
+    os.environ["MASTER_PORT"] = "29500"
+    # os.environ["TORCH_DISTRIBUTED_DEBUG"] = "DETAIL"
+    # os.environ['TP_SOCKET_IFNAME']='lo'
+    options = rpc.TensorPipeRpcBackendOptions(
+        num_worker_threads=16,
+        init_method=f"tcp://{rank0_ip}:10003",
+        rpc_timeout=MAX_TIME_TO_CONNECT,
+        _transports=["uv"],
+        # Currently fails when nodes have more than 0 gpus avail,
+        # even when no device is made visible
+        devices=[],
+    )
+    print("init rpc")
+    rpc.init_rpc(
+        f"COLLECTOR_NODE_{rank}",
+        rank=rank,
+        backend=rpc.BackendType.TENSORPIPE,
+        rpc_backend_options=options,
+        world_size=world_size,
+    )
+    print("waiting...")
+    rpc.shutdown()
+
+
+class DistributedDataCollector(_DataCollector):
+    def __init__(
+        self,
+        env_makers,
+        policy,
```

Comment:
won't policy by in collector_kwargs as well? is there any case in which it will be different?